packages/types/src/integration.ts

```diff
@@ -1,8 +1,11 @@
 import { EventProcessor } from './eventprocessor';
 import { Hub } from './hub';
 
-/** Integration Class Interface */
-export interface IntegrationClass<T> {
+/**
+ * Integration class interface. Note that this is separate from the {@link Integration} interface, which is for
+ * instances of the class described by this interface.
+ */
+export interface IntegrationClass<T = Integration> {
```

Comment:
I would like to see a more generic `ClassConstructor<T>` type:

```ts
export interface ClassConstructor<T> {
  new (...args: any[]): T;
}

export interface IntegrationClass extends ClassConstructor<Integration> {
  id: string;
}

export interface Integration {
  name: string;
}

class BrowserTracing implements Integration {
  public static id: string = 'BrowserTracing';

  public name: string = BrowserTracing.id;
}

const bt: IntegrationClass = BrowserTracing;

const inst: Integration = new bt();
```

Reply:
I might name it slightly differently, but I'm not against it _if_ this is a problem we're trying to solve in more than just this once instance. Regardless, I think that should be a different PR, because it's a broader scope than this change.

Reply:
I like the inclusion of `ClassConstructor` here because it makes the difference between `Integration` and `IntegrationClass` much more explicit. Makes it easier for future contributors to understand (otherwise I can see other's having the convo like we did in that slack thread). I don't think making `IntegrationClass` extend another type broadens the scope that much.

I see the future PR being we move `ClassConstructor` out of this file and start typing other classes vs. instances.